datasets/flwr_datasets/partitioner/vertical_size_partitioner.py

```diff
@@ -0,0 +1,300 @@
+
+    Parameters
+    ----------
+    partition_sizes : Union[list[int], list[float]]
+        A list where each value represents the size of a partition.
+        list[int] -> each value represent an absolute number of columns. Size zero is
+        allowed and will result in an empty partition if no shared columns are present.
+        list of floats -> each value represent a fraction total number of columns.
+        Note that applies to collums without `active_party_columns` or `shared_columns`.
+        They are additionally included in to the partition(s).
+    active_party_column : Optional[Union[str, list[str]]]
+        Column(s) (typically representing labels) associated with the
+        "active party" (which can be the server).
+    active_party_columns_mode : Union[Literal[["add_to_first", "add_to_last", "create_as_first", "create_as_last", "add_to_all"], int]
+        Determines how to assign the active party columns:
+        - "add_to_first": Append active party columns to the first partition.
+        - "add_to_last": Append active party columns to the last partition.
+        - int: Append active party columns to the specified partition index.
+        - "create_as_first": Create a new partition at the start containing only
+            these columns.
+        - "create_as_last": Create a new partition at the end containing only
+            these columns.
+        - "add_to_all": Append active party columns to all partitions.
+    drop_columns : Optional[list[str]]
+        Columns to remove entirely from the dataset before partitioning.
+    shared_columns : Optional[list[str]]
+        Columns to duplicate into every partition after initial partitioning.
+    shuffle : bool
+        Whether to shuffle the order of columns before partitioning.
+    seed : Optional[int]
+        Random seed for shuffling columns. Has no effect if `shuffle=False`.
+
+    Examples
+    --------
+    >>> from flwr_datasets import FederatedDataset
+    >>> from flwr_datasets.partitioner import VerticalSizePartitioner
+    >>>
+    >>> partitioner = VerticalSizePartitioner(
+    ...     partition_sizes=[8, 4, 2],
+    ...     active_party_column="income",
+    ...     active_party_columns_mode="create_as_last"
+    ... )
+    >>> fds = FederatedDataset(
+    ...     dataset="scikit-learn/adult-census-income",
+    ...     partitioners={"train": partitioner}
+    ... )
+    >>> partitions = [fds.load_partition(i) for i in range(fds.partitioners["train"].num_partitions)]
+    >>> print([partition.column_names for partition in partitions])
+    """
+
+    def __init__(
+        self,
+        partition_sizes: Union[list[int], list[float]],
+        active_party_column: Optional[Union[str, list[str]]] = None,
+        active_party_columns_mode: Union[
+            Literal[
+                "add_to_first",
+                "add_to_last",
+                "create_as_first",
+                "create_as_last",
+                "add_to_all",
+            ],
```

Comment:
it might be worth defining this at the top of the file so to avoid redefining it later a couple of times (see `_validate_parameters_in_init()` and `vertical_partitioner_utils.py`)

Reply:
I can't find it in Python docs exactly but `Literal[*some_list]` gives a syntax error (at least for Python 3.10 so it should be the problem at least up to that point)